app/src/main/java/com/hedvig/app/feature/genericauth/EmailInputScreen.kt

```diff
@@ -0,0 +1,179 @@
+package com.hedvig.app.feature.genericauth
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.OutlinedTextField
+import androidx.compose.material.Text
+import androidx.compose.material.TopAppBar
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material.icons.filled.Clear
+import androidx.compose.material.icons.outlined.ErrorOutline
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.onFocusChanged
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.google.accompanist.insets.navigationBarsWithImePadding
+import com.google.accompanist.insets.systemBarsPadding
+import com.hedvig.app.R
+import com.hedvig.app.ui.compose.composables.buttons.LargeContainedButton
+import com.hedvig.app.ui.compose.theme.HedvigTheme
+import com.hedvig.app.util.compose.submitOnEnter
+
+@Composable
+fun EmailInputScreen(
+    onUpClick: () -> Unit,
+    onInputChanged: (String) -> Unit,
+    onSubmitEmail: () -> Unit,
+    onClear: () -> Unit,
+    onBlur: () -> Unit,
+    inputValue: String,
+    error: GenericAuthViewModel.ViewState.TextFieldError?,
```

Comment:
Pass string here instead of ViewState? To avoid coupling this screen to GenericAuthViewModel

Reply:
Definitely a good idea about not passing down the GraphQL type all the way down here. And as far as the dialog goes, probably Hiro will have an opinion right?

Reply:
I guess we could do the error -> string-conversion at the calling layer? Having the parent `Activity` be the one dependent on the `ViewModel`. I'll make that change!